### PR TITLE
Clean up the timing configuration messages.

### PR DESCRIPTION
**Pull request by nomis52**
- Give them sensible names
- Add a param for configuring the responder delay
- Add a bunch of unit tests

### CODE REVIEW

File: firmware/src/constants.h [hunk 1]
```diff
@@ -125,40 +125,86 @@ typedef enum {
...
    */
-  GET_MAB_TIME = 0x13,
+  COMMAND_GET_MARK_TIME = 0x13,
 
   // Advanced Configuration
   /**
-   * @brief Set the RDM Broadcast listen time.
-   * See @ref message_cmd_setrdmbcastlisten.
+   * @brief Set the RDM Broadcast timeout.
+   * See @ref message_cmd_setrdmbcasttimeout.
```
Comment (peternewman) [inline firmware/src/constants.h]: is there any reason this isn't called broadcast?

File: firmware/src/constants.h [hunk 2]
```diff
@@ -244,23 +290,31 @@ typedef enum {
...
  * Measured is in 10ths of a millisecond, from Line 1 & 3, Table 3-2, E1.20.
  * Responders have 2ms (see Table 3-4), and then in line proxies can introduce
  * up to 704uS of delay. This rounds to 2.8 ms.
  */
-#define DEFAULT_RDM_WAIT_TIME 28
+#define DEFAULT_RDM_RESPONSE_TIMEOUT 28
 
 /**
- * @brief The default time a RDM DUB response can take.
+ * @brief The default maximum time a RDM DUB response can take.
```
Comment (peternewman) [inline firmware/src/constants.h]: SPaG an

File: firmware/src/constants.h (hunk 1 above)
Comment (nomis52) [inline firmware/src/constants.h]: The refs names are becoming unwieldily. I tried to shorten them.

File: firmware/src/transceiver.c [hunk 3]
```diff
@@ -44,6 +44,16 @@
 
 #define INPUT_CAPTURE_MODULE IC_ID_2
 
+/*
+ * @brief The minimum RDM responder delay in 10ths of a micro, Table 3-4, E1.20
```
Comment (jmoyerman) [inline firmware/src/transceiver.c]: Picky: microsecond?

File: firmware/src/transceiver.c [hunk 4]
```diff
@@ -44,6 +44,16 @@
...
+ */
+#define MINIMUM_RESPONDER_DELAY 1760
+
+/*
+ * @brief The maximum RDM responder delay in 10ths of a micro. Table 3-4, E1.20
```
Comment (jmoyerman) [inline firmware/src/transceiver.c]: as above

Comment (nomis52) [inline firmware/src/transceiver.c]: Fixed

File: firmware/src/transceiver.c (hunk 3 above)
Comment (nomis52) [inline firmware/src/transceiver.c]: Fixed

File: firmware/src/constants.h (hunk 2 above)
Comment (nomis52) [inline firmware/src/constants.h]: Fixed

File: firmware/src/message_handler.c [hunk 5]
```diff
@@ -95,66 +101,170 @@ static void SetMarkTime(uint8_t token,
...
+    uint8_t token,
+    const uint8_t* payload,
+    unsigned int length) {
+  uint16_t timeout;
+  if (length != sizeof(timeout)) {
+    SendMessage(token, COMMAND_SET_RDM_RESPONSE_TIMEOUT, RC_BAD_PARAM, NULL, 0);
+    return;
+  }
+
+  timeout = payload[0] + (payload[1] << 8);
```
Comment (peternewman) [inline firmware/src/message_handler.c]: JoinUInt16

File: firmware/src/message_handler.c [hunk 6]
```diff
@@ -95,66 +101,170 @@ static void SetMarkTime(uint8_t token,
...
+  if (length != sizeof(limit)) {
+    SendMessage(token, COMMAND_SET_RDM_DUB_RESPONSE_LIMIT, RC_BAD_PARAM, NULL,
+                0);
     return;
   }
 
-  wait_time = payload[0] + (payload[1] << 8);
-  bool ok = Transceiver_SetRDMWaitTime(wait_time);
-  SendMessage(token, SET_RDM_WAIT_TIME, ok ? RC_OK : RC_BAD_PARAM, NULL, 0);
+  limit = payload[0] + (payload[1] << 8);
```
Comment (peternewman) [inline firmware/src/message_handler.c]: JoinUInt16

File: firmware/src/message_handler.c [hunk 7]
```diff
@@ -95,66 +101,170 @@ static void SetMarkTime(uint8_t token,
...
+static void SetRDMResponderDelay(uint8_t token,
+                                 const uint8_t* payload,
+                                 unsigned int length) {
+  uint16_t delay;
+  if (length != sizeof(delay)) {
+    SendMessage(token, COMMAND_SET_RDM_RESPONDER_DELAY, RC_BAD_PARAM, NULL, 0);
+    return;
+  }
+
+  delay = payload[0] + (payload[1] << 8);
```
Comment (peternewman) [inline firmware/src/message_handler.c]: Again

File: firmware/src/message_handler.c [hunk 8]
```diff
@@ -95,66 +101,170 @@ static void SetMarkTime(uint8_t token,
...
+static void SetRDMResponderJitter(uint8_t token,
+                                  const uint8_t* payload,
+                                  unsigned int length) {
+  uint16_t jitter;
+  if (length != sizeof(jitter)) {
+    SendMessage(token, COMMAND_SET_RDM_RESPONDER_JITTER, RC_BAD_PARAM, NULL, 0);
+    return;
+  }
+
+  jitter = payload[0] + (payload[1] << 8);
```
Comment (peternewman) [inline firmware/src/message_handler.c]: Again

File: firmware/src/message_handler.c (hunk 5 above)
Comment (nomis52) [inline firmware/src/message_handler.c]: Done for all.

File: firmware/src/constants.h (hunk 1 above)
Comment (peternewman) [inline firmware/src/constants.h]: Okay.

**Comment by peternewman** [comment]
Can you fix the other outstanding comments please?

**Comment by nomis52** [comment]
I think I got all of them, which ones did I miss?

**Pull request merged**